pkgs/applications/audio/musescore/default.nix

```diff
@@ -1,46 +1,39 @@
-{ mkDerivation, lib, fetchFromGitHub, cmake, pkg-config
-, alsa-lib, freetype, libjack2, lame, libogg, libpulseaudio, libsndfile, libvorbis
-, portaudio, portmidi, qtbase, qtdeclarative, qtgraphicaleffects
-, qtquickcontrols2, qtscript, qtsvg, qttools
-, qtwebengine, qtxmlpatterns
+{ stdenv, lib, fetchFromGitHub, cmake, pkg-config
+, libjack2, libpulseaudio
+, libvorbis, libogg
+, libsndfile, alsa-lib, openssl_1_1, lame
+, qtdeclarative, qtgraphicaleffects, qtscript, qtsvg, qtwebengine
+, qtnetworkauth, qtquickcontrols2, qtxmlpatterns, qttools
+, qtx11extras
+, wrapQtAppsHook
+, autoPatchelfHook
 , nixosTests
 }:
 
-mkDerivation rec {
+stdenv.mkDerivation rec {
   pname = "musescore";
-  version = "3.6.2";
+  version = "4.0";
 
   src = fetchFromGitHub {
     owner = "musescore";
     repo = "MuseScore";
     rev = "v${version}";
-    sha256 = "sha256-GBGAD/qdOhoNfDzI+O0EiKgeb86GFJxpci35T6tZ+2s=";
+    sha256 = "sha256-mc3FYLrzEyUHNFH/P57dWKKHl2Lxew1AQBu+70zmLJs=";
   };
 
-  patches = [
-    ./remove_qtwebengine_install_hack.patch
-  ];
+  nativeBuildInputs = [ cmake pkg-config ];
 
   cmakeFlags = [
     "-DMUSESCORE_BUILD_CONFIG=release"
-    "-DUSE_SYSTEM_FREETYPE=ON"
-  ];
-
-  qtWrapperArgs = [
-    # MuseScore JACK backend loads libjack at runtime.
-    "--prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ libjack2 ]}"
-    # There are some issues with using the wayland backend, see:
-    # https://musescore.org/en/node/321936
-    "--set-default QT_QPA_PLATFORM xcb"
   ];
 
-  nativeBuildInputs = [ cmake pkg-config ];
-
   buildInputs = [
-    alsa-lib libjack2 freetype lame libogg libpulseaudio libsndfile libvorbis
-    portaudio portmidi # tesseract
-    qtbase qtdeclarative qtgraphicaleffects qtquickcontrols2
-    qtscript qtsvg qttools qtwebengine qtxmlpatterns
+    libjack2 libpulseaudio libvorbis libogg
+    lame libsndfile alsa-lib openssl_1_1
+    qtnetworkauth qtquickcontrols2 qtxmlpatterns qttools
+    qtx11extras qtdeclarative qtgraphicaleffects qtscript
+    qtsvg qtwebengine
+    autoPatchelfHook wrapQtAppsHook
```

Comment:
`autoPatchelfHook` and `wrapQtAppsHook` should be in `nativeBuildInputs`? Also, why did you switch from `mkDerivation` to `stdenv.mkDerivation` and manually insert `wrapQtAppsHook`?

Regarding patching binaries, upstream should be completely open source, hence I think we should track down what binaries are not compiled and why, and fix that (send patches upstream if needed).

Reply:
> Also, why did you switch from `mkDerivation` to `stdenv.mkDerivation` and manually insert `wrapQtAppsHook`?

FTR this does align with the [currently recommended convention](https://nixos.org/manual/nixpkgs/stable/#qt-default-nix) as of  02924cf9514f1bb4cf8d82f80fba64c830afe54d.

Reply:
> > Also, why did you switch from `mkDerivation` to `stdenv.mkDerivation` and manually insert `wrapQtAppsHook`?
> 
> FTR this does align with the [currently recommended convention](https://nixos.org/manual/nixpkgs/stable/#qt-default-nix) as of [02924cf](https://github.com/NixOS/nixpkgs/commit/02924cf9514f1bb4cf8d82f80fba64c830afe54d).

The manual doesn't specifically says it's preferred to use `stdenv.mkDerivation` with `wrapQtAppsHook` over `mkDerivation`. I assume that part of the docs was written before the wrapper `mkDerivation` function was created. Otherwise, what would have been the point in defining `mkDerivation`?

Reply:
The commit I linked above, 02924cf9514f1bb4cf8d82f80fba64c830afe54d, is from January 2021, and *removed* a previous recommendation to use `mkDerivation` over `stdenv.mkDerivation`. See also

- #180841

Reply:
> The commit I linked above, [02924cf](https://github.com/NixOS/nixpkgs/commit/02924cf9514f1bb4cf8d82f80fba64c830afe54d), is from January 2021, and _removed_ a previous recommendation to use `mkDerivation` over `stdenv.mkDerivation`. See also
> 
>     * [Qt `mkDerivation` is slightly broken, redundant and should be deprecated #180841](https://github.com/NixOS/nixpkgs/issues/180841)

OK Good to know, thanks for the links :).

Still, `wrapQtAppsHook` should be in `nativeBuildInputs`. and ideally `autoPatchelfHook`.should not be used.

Reply:
`autoPatchelfHook` and `wrapQtAppsHook` have been moved to `nativeBuildInputs`. I tried using `preFixup` and `patchelf` directly but it did not work.

Reply:
Also I used `wrapQtAppsHook` instead of `libsForQt5.mkDerivation` by accident, not because of best practices.

Reply:
Replying to the comment at my PR (https://github.com/NixOS/nixpkgs/pull/206559#issuecomment-1356816813):

> Is `autoPatchelfHook` something bad? Also, I tried with `openssl` and it didn’t work, only `openssl_1_1` worked…

`autoPatchelfHook` is not "bad", it's just that this project shouldn't need it - From what I understand (from the [docs](https://nixos.org/manual/nixpkgs/unstable/#setup-hook-autopatchelfhook) and from experience), it should be used only in cases were we declare for a justified reasons that we do not strive to compile the software we package - usually for proprietary software. `musescore` is completely free software, and Google's crashpad, bundled in musescore's source, that needs from some reason to be patched with this hook, is also free software. Hence I don't see a reason to use this hook, and I think we should make the effort to fix the musescore build, such that it will not require this hook.

Reply:
I see.